### PR TITLE
Refactofing and Bug fixes

### PR DESCRIPTION
**Pull request by ManhLeDuc**
## Related Tickets
- [#40891](https://edu-redmine.sun-asterisk.vn/issues/40891)

## WHAT (optional)
- Refactofing and Bug fixes

## HOW
- Refactoring supervisor course view for trainee course view.
- Fix bugs.

## WHY (optional)
- Better code reuse

## Evidence (Screenshot or Video)

### CODE REVIEW

**Comment by trannguyen61** [comment]
OK

**Comment by nguyenquanganhc** [comment]
OK

**Comment by ManhLeDuc** [comment]
ready

File: app/views/enrollments/_enrollment.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<tr class="subjects">
+  <td><%= enrollment_counter + 1%></td>
```
Comment (ghost) [inline app/views/enrollments/_enrollment.html.erb]:
```suggestion
  <td><%= enrollment_counter + 1 %></td>
```

File: app/views/enrollments/_enrollment.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <td><%= enrollment.user_name %></td>
+  <td><%= enrollment.user_email %></td>
+  <td>
+    <% if current_user.role_supervisor? %>
+      <%= link_to enrollment, method: :delete, remote: true, data: {confirm: t("delete_confirm")},
+      class: "text-danger", "data-mdb-toggle":"tooltip", title: t("delete") do %>
+        <i class="fa fa-trash-alt me-3"></i>
+      <% end %>
+    <% end %>
+    <%= link_to enrollment, class: "text-info", "data-mdb-toggle":"tooltip", title: t("details") do %>
```
Comment (ghost) [inline app/views/enrollments/_enrollment.html.erb]:
```suggestion
    <%= link_to enrollment, class: "text-info", data: {mdb-toggle: "tooltip"}, title: t("details") do %>
```
Thử thế này xem em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ManhLeDuc** [comment]
ready

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+
+10.times do |n|
+  name = "giaovien#{n + 1}"
+  email = "giaovien#{n + 1}@test.com"
+  password = "123123"
+  User.create!  name: name,
+                email: email,
+                password: password,
+                password_confirmation: password,
+                role: 1
```
Comment (ngocvt-0484) [inline db/seeds.rb]: 1 là gì ?
k baoh dc goi số ntn e nhé

File: db/seeds.rb [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
-#   Character.create(name: 'Luke', movie: movies.first)
+10.times do |n|
+  name = "hocsinh#{n + 1}"
+  email = "hocsinh#{n + 1}@test.com"
+  password = "123123"
+  User.create!  name: name,
+                email: email,
+                password: password,
+                password_confirmation: password,
+                role: 0
```
Comment (ngocvt-0484) [inline db/seeds.rb]: same

**Comment by ManhLeDuc** [comment]
ready

**Pull request merged**